Source/Core/Core/HW/DSPHLE/UCodes/AESnd.cpp

```diff
@@ -0,0 +1,403 @@
+    // This applies throughout.
+    u32 delay = m_parameter_block.delay;
+    u32 sample_index = 0;
+    u32 remaining_samples = NUM_OUTPUT_SAMPLES;
+
+    if (delay != 0)
+    {
+      // delay_loop
+      // This is implemented with a BLOOP and then a read from the loop counter stack register in
+      // the actual uCode. I'm not 100% sure how it works/if it works correctly there.
+      while (remaining_samples != 0)
+      {
+        remaining_samples--;
+        delay--;
+        if (delay == 0)
+          break;
+        sample_index++;
+      }
+      // exit_delay
+      m_parameter_block.delay = delay;
+    }
+    // no_delay - operates in set40 mode
+    while (remaining_samples != 0)  // BLOOP dspmixer_loop_end
+    {
+      remaining_samples--;
+      s32 right = m_output_buffer[sample_index * 2 + 0];
+      s32 left = m_output_buffer[sample_index * 2 + 1];
+      right += m_parameter_block.right;
+      left += m_parameter_block.left;
+      // Clamping from set40 mode
+      right = std::clamp(right, -32768, 32767);
+      left = std::clamp(left, -32768, 32767);
+      m_output_buffer[sample_index * 2 + 0] = right;
+      m_output_buffer[sample_index * 2 + 1] = left;
+      sample_index++;
+
+      const u32 counter = static_cast<u32>(m_parameter_block.counter) + m_parameter_block.freq;
+      m_parameter_block.counter = static_cast<u16>(counter);
+      u16 counter_h = counter >> 16;
+      if (counter_h >= 1)
+      {
+        s16 new_l = 0, new_r = 0;
+
+        // jrge $ar3 (using select_mixer table)
+        // We already masked voice_format with the right mask for the version, so nothing special
+        // needs to be done for this switch statement to exclude unsupported formats
+        switch (voice_format)
+        {
+        case VOICE_MONO8:
+        case VOICE_MONO16:
+          // mono_mix
+          while (counter_h >= 1)
+          {
+            counter_h--;
+            new_r = s_accelerator->Read(ACCELERATOR_COEFS.data());
+            new_l = new_r;
+          }
+          break;
+
+        case VOICE_STEREO8:
+        case VOICE_STEREO16:
+          // stereo_mix
+          while (counter_h >= 1)
+          {
+            counter_h--;
+            new_r = s_accelerator->Read(ACCELERATOR_COEFS.data());
+            new_l = s_accelerator->Read(ACCELERATOR_COEFS.data());
+          }
+          break;  // falls through to mix_samples normally
+
+        case VOICE_MONO8_UNSIGNED:
+        case VOICE_MONO16_UNSIGNED:
+          // mono_unsigned_mix
+          while (counter_h >= 1)
+          {
+            counter_h--;
+            new_r = s_accelerator->Read(ACCELERATOR_COEFS.data());
+            new_l = new_r;
+          }
+          new_r ^= 0x8000;
+          new_l ^= 0x8000;
+          break;
+
+        case VOICE_STEREO8_UNSIGNED:
+        case VOICE_STEREO16_UNSIGNED:
+          // stereo_unsigned_mix
+          while (counter_h >= counter_h)
```

Comment:
Shouldn't this be `while (counter_h >= 1)` or am I missing something?

Reply:
Yeah, that's definitely wrong. I think that happened because I originally had `while (counter >= 0x10000)`, and I changed that one incorrectly.